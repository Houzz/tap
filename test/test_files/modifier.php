<?php

class A {
    public $publicProp;
    private $privateProp;
    protected $protectedProp;
    var $defaultProp;

    function defaultMethod() {
        $this->privateA();
        $this->privateProp = 1;
    }

    public function publicA() {
    }

    private function privateA() {
    }

    protected function protectedA() {
    }
}

class B extends A {
    public function publicB() {
        $this->privateB();
        $this->privateA();  // wrong
        $this->protectedA();
        $this->publicA();
        $this->publicC();   // wrong
    }
    private function privateB() {}
}

class C {
    public function publicC() {}
}

$a = new A();
echo $a->publicProp;
echo $a->protectedProp; // wrong
echo $a->privateProp;   // wrong
echo $a->defaultProp;
$a->publicA();
$a->privateA();     // wrong
$a->protectedA();   // wrong
$a->defaultMethod();

$b = new B();
$b->publicB();
$b->publicA();
$b->publicC();      // wrong
$b->privateB();     // wrong

class D extends B {
    public function publicD() {
        $this->protectedA();
        $this->privateA();  // wrong
    }
}
