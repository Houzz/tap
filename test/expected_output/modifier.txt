27 E_METHOD_NOT_FOUND
30 E_METHOD_NOT_FOUND
41 E_PROPERTY_NOT_FOUND
42 E_PROPERTY_NOT_FOUND
45 E_METHOD_NOT_FOUND
46 E_METHOD_NOT_FOUND
52 E_METHOD_NOT_FOUND
53 E_METHOD_NOT_FOUND
58 E_METHOD_NOT_FOUND
